### ISSUE TITLE
Change vs. Insert/Remove operations

### ISSUE THREAD

**Issue by naterush** [saga-vcs/saga#39]
We have the following object `[[1, 2, 3]]` that changes into `[[1, 2, 3, 4]]`. What operation was performed here? Was this a change of the path `[0]` from `[1, 2, 3]` to `[1, 2, 3, 4]`, or was this an insert on path `[0, 3]`, or was it both? 

I'm leaning towards defining change operations only on "leaf elements" of the type tree (e.g. this would be an insert), and then 

In general, we need to more thoroughly investigate what an operation is + what set we want to support at a framework level!

**Comment by aarondr77** [comment]
I agree with your initial hunch for this example. In this case, I think it would make more sense to characterize it as an insert. 

Is this always true though? It might be file type specific. I wonder if this is something that we should allow the developers to choose when specifying a new file type. I could imagine that in Excel you would always want the changes at the leaf node. But perhaps in other file types, where the base unit, U, that the user actually cares about is a more complex type (ie: A list with dictionaries) where that specific configuration represents one note, we would actually care that the entire unit U was changed to U' - not that there was an insert in a dictionary.
